pkg/modules/dnsnames.go

```diff
@@ -0,0 +1,80 @@
+package modules
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-ldap/ldap/v3"
+	"github.com/ropnop/go-windapsearch/pkg/ldapsession"
+	"github.com/spf13/pflag"
+)
+
+type DnsNamesModule struct{}
+
+func init() {
+	AllModules = append(AllModules, new(DnsNamesModule))
+}
+
+func (d DnsNamesModule) Name() string {
+	return "dns-names"
+}
+
+func (d DnsNamesModule) Description() string {
+	return "List all DNS Names"
+}
+
+func (d DnsNamesModule) FlagSet() *pflag.FlagSet {
+	flags := pflag.NewFlagSet(d.Name(), pflag.ExitOnError)
+	return flags
+}
+
+func (d DnsNamesModule) DefaultAttrs() []string {
+	return []string{"dn"}
+}
+
+func (d DnsNamesModule) Run(session *ldapsession.LDAPSession, attrs []string) error {
+	locations := []string{"CN=MicrosoftDNS,DC=DomainDnsZones,%s", "CN=MicrosoftDNS,DC=ForestDnsZones,%s", "CN=MicrosoftDNS,CN=System,%s"}
+
+	// I'm too stupid for filtering these entries via LDAP, tried for about a day and failed... now they're implemented here, feel free to improve this
```

Comment:
Lol, ldap filters are the worst. I'm gonna explore and see if there's a way to do it in LDAP syntax as well but I'll probably burn a day and end up accepting this anyway :)

Reply:
I've read several threads about Microsoft AD not implementing filters for the DN, but I'm not sure if this is correct and the actual issue here.

Reply:
Yeah filtering isn't gonna work here. I've been reading through Dirkjan's code and the only filtering he does is in the CSV writing phase. I'm going to explore moving this logic to the marshaller and results writer instead of inside the module itself